src/d3d8/d3d8_device.cpp

```diff
@@ -214,14 +212,16 @@ namespace dxvk {
     // Resetting implicitly ends scenes started by BeginScene
     GetD3D9()->EndScene();
 
+    m_presentParams = *pPresentationParameters;
+    ResetState();
+
     d3d9::D3DPRESENT_PARAMETERS params = ConvertPresentParameters9(pPresentationParameters);
     HRESULT res = GetD3D9()->Reset(&params);
 
     if (FAILED(res))
       return res;
 
-    m_presentParams = *pPresentationParameters;
-    ResetState();
+    RegenBackBuffersAndAutoDepthStencil();
```

Comment:
Recreate sounds nicer. :)

Reply:
I'm very uninspired when it comes to naming...